From 68876ccdd2fdb5bf450d9d546e18286b4508cf33 Mon Sep 17 00:00:00 2001
Message-Id: <68876ccdd2fdb5bf450d9d546e18286b4508cf33.1561531622.git.aditya@kobol.io>
From: Aditya Prayoga <aditya@kobol.io>
Date: Wed, 26 Jun 2019 14:42:30 +0800
Subject: [PATCH] mvebu: configs: add boot-marvell.cmd backward compatibility

On system that still use boot.scr derived from boot-marvell.cmd, new
u-boot 2018 will failed to load dtb and script due to missing some
variables. This will render the system unbootable.

These changes added the missing variables.

Signed-off-by: Aditya Prayoga <aditya@kobol.io>
---
 include/config_distro_bootcmd.h | 1 +
 include/configs/clearfog.h      | 2 ++
 include/configs/helios4.h       | 2 ++
 3 files changed, 5 insertions(+)

diff --git a/include/config_distro_bootcmd.h b/include/config_distro_bootcmd.h
index 4993303f..97af842e 100644
--- a/include/config_distro_bootcmd.h
+++ b/include/config_distro_bootcmd.h
@@ -411,6 +411,7 @@
 	"boot_a_script="                                                  \
 		"load ${devtype} ${devnum}:${distro_bootpart} "           \
 			"${scriptaddr} ${prefix}${script}; "              \
+		"setenv boot_interface ${devtype};"                       \
 		"source ${scriptaddr}\0"                                  \
 	\
 	"scan_dev_for_scripts="                                           \
diff --git a/include/configs/clearfog.h b/include/configs/clearfog.h
index c51cf284..bbbfff71 100644
--- a/include/configs/clearfog.h
+++ b/include/configs/clearfog.h
@@ -148,6 +148,8 @@
 	LOAD_ADDRESS_ENV_SETTINGS \
 	"fdtfile=" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
 	"console=ttyS0,115200\0" \
+	"loadaddr=0x02000000\0" \
+	"fdt_addr=" FDT_ADDR_R "\0" \
 	BOOTENV
 
 #endif /* CONFIG_SPL_BUILD */
diff --git a/include/configs/helios4.h b/include/configs/helios4.h
index df4d8bcb..ef6c4ed6 100644
--- a/include/configs/helios4.h
+++ b/include/configs/helios4.h
@@ -174,6 +174,8 @@
 	LOAD_ADDRESS_ENV_SETTINGS \
 	"fdtfile=" CONFIG_DEFAULT_DEVICE_TREE ".dtb\0" \
 	"console=ttyS0,115200\0" \
+	"loadaddr=0x02000000\0" \
+	"fdt_addr=" FDT_ADDR_R "\0" \
 	BOOTENV
 
 #endif /* CONFIG_SPL_BUILD */
-- 
2.17.1

